File: app/src/main/java/com/daniel/mobilepauker2/model/TextDrawable.java
package com.daniel.mobilepauker2.model;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.graphics.Typeface;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.OvalShape;
import android.support.annotation.NonNull;

public class TextDrawable extends ShapeDrawable {
    private String s;
    private int size = 120;

    public TextDrawable(String text) {
        this(text, Color.BLACK);
    }

    public TextDrawable(int backColor) {
        this("", backColor);
    }

    private TextDrawable(String text, int backColor) {
        super(new OvalShape());
        s = text;
        getPaint().setColor(backColor);
        setIntrinsicHeight(size);
        setIntrinsicWidth(size);
        setBounds(0, 0, size, size);
    }

    @Override
    public void draw(@NonNull Canvas canvas) {
        super.draw(canvas);

        Paint paint = getPaint();
        if (!s.isEmpty()) {
            paint.setColor(Color.WHITE);
            paint.setTextSize(size / 2);
            Rect bounds = new Rect();
            paint.getTextBounds(s, 0, s.length(), bounds);
            int theight = bounds.height();
            int twidth = bounds.width();
            canvas.drawText(s, (size / 2) - (twidth / 2), (size / 2) + (theight / 2), paint);
        } else {
            paint.setColor(Color.BLACK);
            paint.setStyle(Paint.Style.STROKE);
            canvas.drawCircle(size / 2, size / 2, size / 2, paint);
        }
    }

    public void setBold(boolean bool) {
        int bold = bool ? Typeface.BOLD : Typeface.NORMAL;
        getPaint().setTypeface(Typeface.create(Typeface.DEFAULT, bold));
    }

    public void setItalic(boolean bool) {
        int italic = bool ? Typeface.ITALIC : Typeface.NORMAL;
        getPaint().setTypeface(Typeface.create(Typeface.DEFAULT, italic));
    }
}

File: app/src/main/java/com/daniel/mobilepauker2/core/model/ui/TextDrawable.kt
package com.daniel.mobilepauker2.core.model.ui

import android.graphics.*
import android.graphics.drawable.ShapeDrawable
import android.graphics.drawable.shapes.OvalShape

class TextDrawable private constructor(private val s: String, backColor: Int) :
    ShapeDrawable(OvalShape()) {
    private val size = 120

    constructor(text: String) : this(text, Color.BLACK)

    constructor(backColor: Int) : this("", backColor)

    override fun draw(canvas: Canvas) {
        super.draw(canvas)
        val paint = paint
        if (s.isNotEmpty()) {
            paint.color = Color.WHITE
            paint.textSize = size / 2.toFloat()
            val bounds = Rect()
            paint.getTextBounds(s, 0, s.length, bounds)
            val theight = bounds.height()
            val twidth = bounds.width()
            canvas.drawText(
                s,
                size / 2 - (twidth / 2).toFloat(),
                size / 2 + (theight / 2).toFloat(),
                paint
            )
        } else {
            paint.color = Color.BLACK
            paint.style = Paint.Style.STROKE
            canvas.drawCircle(size / 2.toFloat(), size / 2.toFloat(), size / 2.toFloat(), paint)
        }
    }

    fun setBold(bool: Boolean) {
        val bold = if (bool) Typeface.BOLD else Typeface.NORMAL
        paint.typeface = Typeface.create(Typeface.DEFAULT, bold)
    }

    fun setItalic(bool: Boolean) {
        val italic = if (bool) Typeface.ITALIC else Typeface.NORMAL
        paint.typeface = Typeface.create(Typeface.DEFAULT, italic)
    }

    init {
        paint.color = backColor
        intrinsicHeight = size
        intrinsicWidth = size
        setBounds(0, 0, size, size)
    }
}